lib/logstash/inputs/udp.rb

```diff
@@ -47,8 +47,6 @@ def run(output_queue)
     begin
       # udp server
       udp_listener(output_queue)
-    rescue LogStash::ShutdownSignal
-      # do nothing, shutdown was requested.
     rescue => e
       @logger.warn("UDP listener died", :exception => e, :backtrace => e.backtrace)
       sleep(5)
```

Comment:
What happen if you are sleeping here? I think you might need to catch this situation and do a wake up, isn't?